app/controllers/sessions_controller.rb

```diff
@@ -14,6 +14,7 @@ def new
     store_fallback
     referer = request.referer
     session[:from_cnx] = (referer =~ /cnx\.org/) unless referer.blank?
+    session[:client_id] = params[:client_id]
```

Comment:
Storing `client_id` in session so that all phases of sign up / in can know that this user came from this application (so they can know not to require terms if applicable).